src/input/game-pads.js

```diff
@@ -120,24 +120,27 @@ pc.extend(pc, function () {
             }
 
             // update current
-            var pads = this.poll();
-            for (i = 0, l = pads.length; i < l; i++) {
-                this.current[i] = pads[i];
-            }
+            this.poll(this.current);
         },
 
         /**
         * @function
         * @name pc.GamePads#poll
         * @description Poll for the latest data from the gamepad API.
+        * @param {Object[]} An optional array used to save the gamepads mapping instead of returning a new one
         * @returns {Object[]} An array of gamepads and mappings for the model of gamepad that is attached
         * @example
         *   var gamepads = new pc.GamePads();
         *   var pads = gamepads.poll();
         *   // pads[0] = { map: <map>, pad: <pad> }
         */
         poll: function () {
-            var pads = [];
+            var pads;
+            if (arguments.length === 0) {
+                pads = [];
+            } else {
+                pads = arguments[0];
```

Comment:
Given the code on L149 where the pad devices are pushed on to the array, this would mean that pads would just grow in size every frame if a gamepad is plugged in?

Reply:
Good catch. The array returned by `poll` was copied into `current` so I thought to let the caller to clear the array but then I removed the `for` loop and forgot to clear it again in `poll`. Thanks!